scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -1397,16 +1514,20 @@
 (defn create-datomic-scheduler
   [conn driver-atom pending-jobs-atom offer-cache heartbeat-ch offer-incubate-time-ms mea-culpa-failure-limit
    fenzo-max-jobs-considered fenzo-scaleback fenzo-floor-iterations-before-warn fenzo-floor-iterations-before-reset
-   fenzo-fitness-calculator task-constraints executor gpu-enabled? good-enough-fitness framework-id
-   {:keys [match-trigger-chan rank-trigger-chan] :as trigger-chans}]
+   fenzo-fitness-calculator task-constraints gpu-enabled? good-enough-fitness framework-id
+   {:keys [executor-config progress-config]}
+   {:keys [match-trigger-chan progress-aggregator-trigger-chan progress-updater-trigger-chan rank-trigger-chan] :as trigger-chans}]
 
   (persist-mea-culpa-failure-limit! conn mea-culpa-failure-limit)
 
   (let [fenzo (make-fenzo-scheduler driver-atom offer-incubate-time-ms fenzo-fitness-calculator good-enough-fitness)
         [offers-chan resources-atom]
-        (make-offer-handler conn driver-atom fenzo framework-id executor pending-jobs-atom offer-cache fenzo-max-jobs-considered
-                            fenzo-scaleback fenzo-floor-iterations-before-warn fenzo-floor-iterations-before-reset match-trigger-chan)]
+        (make-offer-handler conn driver-atom fenzo framework-id executor-config pending-jobs-atom offer-cache fenzo-max-jobs-considered
+                            fenzo-scaleback fenzo-floor-iterations-before-warn fenzo-floor-iterations-before-reset match-trigger-chan)
+        {:keys [batch-size pending-threshold]} progress-config
+        progress-state-chan (progress-update-transactor progress-updater-trigger-chan batch-size conn)
+        {:keys [progress-aggregator-chan]} (progress-update-aggregator pending-threshold progress-aggregator-trigger-chan progress-state-chan)]
     (start-jobs-prioritizer! conn pending-jobs-atom task-constraints rank-trigger-chan)
     {:scheduler (create-mesos-scheduler framework-id gpu-enabled? conn heartbeat-ch
-                                        fenzo offers-chan match-trigger-chan)
+                                        fenzo offers-chan match-trigger-chan progress-aggregator-chan)
```

Comment:
Instead of passing the channel, pass a function that is *something* like `handle-progress-update` which just enqueues to update. That way we can better localize the logic for handling progress updates. Also, the code `handle-status-update` code shouldn't need to know about the implementation we are using to handle-progress-updates.